// astro-purgecss@4.1.0/dist/index.mjs
import { PurgeCSS } from "purgecss";
import { writeFile } from "node:fs/promises";
import { fileURLToPath } from "node:url";
function handleWindowsPath(outputPath) {
  if (process.platform !== "win32")
    return outputPath;
  if (outputPath.endsWith("\\")) {
    outputPath = outputPath.substring(0, outputPath.length - 1);
  }
  outputPath = outputPath.replaceAll("\\", "/");
  return outputPath;
}
function src_default(options = {}) {
  return {
    name: "astro-purgecss",
    hooks: {
      "astro:build:done": async ({ dir }) => {
        const outDir = handleWindowsPath(fileURLToPath(dir));
        const purged = await new PurgeCSS().purge({
          css: [`${outDir}/**/*.css`],
          defaultExtractor: (content) => content.match(/[\w-/:]+(?<!:)/g) || [],
          ...options,
          content: [
            `${outDir}/**/*.html`,
            `${outDir}/**/*.js`,
            ...options.content || []
          ]
        });
        await Promise.all(
          purged.filter(({ file }) => file == null ? void 0 : file.endsWith(".css")).map(async ({ css, file }) => await writeFile(file, css))
        );
      }
    }
  };
}
export {
  src_default as default
};
